pkg/tenant/db/organizations.go

```diff
@@ -0,0 +1,56 @@
+package db
+
+import (
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/utils/ulids"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const OrganizationNamespace = "organizations"
+
+type Organization struct {
+	ID       ulid.ULID `msgpack:"id"`
+	Name     string    `msgpack:"name"`
+	Created  time.Time `msgpack:"created"`
+	Modified time.Time `msgpack:"modified"`
+}
+
+var _ Model = &Organization{}
```

Comment:
We don't need a whole model for this; what we're going to store is:

- `key`: the serialized ULID of the resource 
- `value`: the serialized ULID of the organization

So instead what we need is basically two functions; a function to retrieve the orgID from the database using the resourceID: 

```go
func GetOrgIndex(resourceID ulid.ULID) (orgID ulid.ULID, err error) {
    var data []byte

    // TODO: add namespace, trtl handling etc. 
    if data, err = db.Get(resourceID[:], nil); err != nil {
        return orgID, err 
    }

    if err = orgID.UnmarshalBinary(data); err != nil {
        return orgID, err
    }
    return orgID, nil
}
```

And a function to store the index pair:

```go
func PutOrgIndex(resourceID, orgID ulid.ULID) error {
    // TODO: add namespace, etc.
    if err = db.Put(resourceID[:], orgID[:], nil); err != nil {
        return err
    }
    return nil
}
```

Then everytime we create a resource; we need to make sure that the above index is correctly saved.

Reply:
The `Get` and `Put` methods in the tenant db take in a context and a model. Should `getRequest` and a `putRequest` be used for `GetOrgIndex` and `PutOrgIndex` and then set the namespace that way or should this be handled differently? https://github.com/rotationalio/ensign/blob/main/pkg/tenant/db/db.go#L168